### ISSUE TITLE
model forward error when input is a randomly generated tensor

### ISSUE THREAD

**Issue by russellfei** [torch/nn#95]
when using randomly generated tensor as an input for network,
it'll assert this kind of error.

``` bash
$ th
> require 'nn'
true    
                                                                      [0.0233s]
> model = nn.Sequential()
                                                                      [0.0001s]
> model:add(nn.SpatialConvolutionMM(3,5,3,3))
nn.Sequential {
  [input -> (1) -> output]
  (1): nn.SpatialConvolutionMM
}
                                                                      [0.0003s]
> img_in = torch.rand(3,20,20):type('torch.FloatTensor')
                                                                      [0.0003s]
> fv_out = model:forward(img_in)
/usr/local/share/lua/5.1/nn/Sequential.lua:37: bad argument #1 (field finput is not a torch.FloatTensor)
stack traceback:
    [C]: in function 'updateOutput'
    /usr/local/share/lua/5.1/nn/Sequential.lua:37: in function 'forward'
    [string "fv_out = model:forward(img_in)"]:1: in main chunk
    [C]: in function 'xpcall'
    /usr/local/share/lua/5.1/trepl/init.lua:696: in function 'repl'
    /usr/local/lib/luarocks/rocks/trepl/scm-1/bin/th:121: in main chunk
    [C]: at 0x00404480  
                                                                      [0.0002s]
```

According to Clement Farabet's `torch/tutorials`, the tensor type could be transferred using `dst = src:type('torch.TypeTensor')` in which `Type` could be 'Float', 'Double', 'Byte', 'Int', etc.
However, this line seems doesn't work at this case.
and if use `print(type(img_in))` we got `userdata`, the same result as the type of `torch.rand(..,..,..)`

In addition, there is one statement way to avoid this error message (using the same model).

``` lua
> temp = torch.rand(3,20,20)
                                                                      [0.0003s]
> temp = temp:type('torch.FloatTensor')
                                                                      [0.0002s]
> img_in = torch.Tensor(3,20,20):copy(temp)
                                                                      [0.0002s]
> fv_out = model:forward(img_in)
```

We can see that the initial types of `temp` and `img_in` are different,
`userdata` for former variable and `torch.FloatTensor` for the latter.

In conclusion, the dataType transfer line `dst = src:Type('..') might be limited or should be revised to stick to its original function.

**Comment by ZENGXH** [comment]
I met with similar problem and found that cast both `model` and the input, ie `img_in` into floattensor help:

```
model = model:float()
img_in = img_in:float()
```
btw, to view the type of object of torch, we can use `torch.type(model)` as nicholas-leonard  mentioned.

although the problem is shown a long time ago but I found that few answers related when I met the bug so it may help for others as an notes..
